proto/zitadel/user/v3alpha/authenticator.proto

```diff
@@ -0,0 +1,468 @@
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629023906488334\"";
+    }
+  ];
+  zitadel.object.v2beta.Details details = 2;
+  // the file type of the key
+  AuthNKeyType type = 3 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"KEY_TYPE_JSON\"";
+    }
+  ];
+  // After the expiration date, the key will no longer be usable for authentication.
+  google.protobuf.Timestamp expiration_date = 4 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"3019-04-01T08:45:00.000000Z\"";
+    }
+  ];
+}
+
+enum AuthNKeyType {
+  AUTHN_KEY_TYPE_UNSPECIFIED = 0;
+  AUTHN_KEY_TYPE_JSON = 1;
+}
+
+message IdentityProvider {
+  // IDP ID is the read-only unique identifier of the identity provider in ZITADEL.
+  string idp_id = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629023906488334\"";
+    }
+  ];
+  // IDP name is the name of the identity provider in ZITADEL.
+  string idp_name = 3 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"google\"";
+    }
+  ];
+  // The user ID represents the ID provided by the identity provider.
+  // This ID is used to link the user in ZITADEL with the identity provider.
+  string user_id = 4 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"as-12-df-89\"";
+    }
+  ];
+  // The username represents the username provided by the identity provider.
+  string username = 5 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"gigi.long-neck@gmail.com\"";
+    }
+  ];
+}
+
+message SetAuthenticators {
+  repeated SetUsername usernames = 1;
+  SetPassword password = 2;
+}
+
+message SetPassword {
+  oneof type {
+    // Provide the plain text password. ZITADEL will take care to store it in a secure way (hash).
+    string password = 1 [
+      (validate.rules).string = {min_len: 1, max_len: 200},
+      (google.api.field_behavior) = REQUIRED,
+      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+        example: "\"Secr3tP4ssw0rd!\"";
+        min_length: 1,
+        max_length: 200;
+      }
+    ];
+    // Encoded hash of a password in Modular Crypt Format:
+    // https://zitadel.com/docs/concepts/architecture/secrets#hashed-secrets.
+    string hash = 2 [
+      (validate.rules).string = {min_len: 1, max_len: 200},
+      (google.api.field_behavior) = REQUIRED,
+      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+        min_length: 1,
+        max_length: 200;
+        example: "\"$2a$12$lJ08fqVr8bFJilRVnDT9QeULI7YW.nT3iwUv6dyg0aCrfm3UY8XR2\"";
+      }
+    ];
+  }
+  // Provide if the user needs to change the password on the next use.
+  bool change_required = 3;
+}
+
+message SendPasswordResetLink {
+  //  NotificationType notification_type = 1; // TODO: did we ever implement this?
```

Comment:
did we ever implement this?

Reply:
Was that an idea like the returnCode for email and phone validation?

Reply:
you should be able to choose whether we send it to your email or phone. if none would be specified it would use the email. unsure if it's really implemented and needed that way

Reply:
Hmm I think currently you can only get a password reset if you have an email verified(through the init code) right? So we would need to add to the logic that it also can be sent to a phone, but as the verification should be indifferent to the medium I guess it should still be included, which could also include a return code.